test/suite/helper_tiered.py

```diff
@@ -260,22 +260,44 @@ def tiered_conn_extensions(self, extlist):
         extlist.extension('storage_sources', self.ss_name + config)
 
     def download_objects(self, bucket_name, prefix):
-        import boto3
-        # The bucket from the storage source is expected to be a name and a region, separated by a 
-        # semi-colon. eg: 'abcd;ap-southeast-2'.
-        bucket_name, region = bucket_name.split(';')
-        
-        # Get the bucket resource and list the objects within that bucket that match the prefix for a
-        # given test.
-        s3 = boto3.resource('s3')
-        bucket = s3.Bucket(bucket_name)
-        objects = list(bucket.objects.filter(Prefix=prefix))
-
         # Create a directory within the test directory to download the objects to.
-        s3_object_files_path = 's3_objects/'
-        if not os.path.exists(s3_object_files_path):
-            os.makedirs(s3_object_files_path)
+        object_files_path = 'objects/'
+        if not os.path.exists(object_files_path):
+            os.makedirs(object_files_path)
+
+        if (self.ss_name == 's3_store'):
+            import boto3
+            # The bucket from the storage source is expected to be a name and a region, separated by a 
+            # semi-colon. eg: 'abcd;ap-southeast-2'.
+            bucket_name, region = bucket_name.split(';')
+            
+            # Get the bucket resource and list the objects within that bucket that match the prefix for a
+            # given test.
+            s3 = boto3.resource('s3')
+            bucket = s3.Bucket(bucket_name)
+            objects = list(bucket.objects.filter(Prefix=prefix))
+
+            for o in objects:
+                file_path = object_files_path + '/' + o.key.split('/')[-1]
+                bucket.download_file(o.key, file_path)
+        elif (self.ss_name == 'gcp_store'):
+            from google.cloud import storage
+            
+            storage_client = storage.Client()
+            bucket = storage_client.bucket(bucket_name)
+            blobs = storage_client.list_blobs(bucket_name, prefix=prefix)
+
+            for blob in blobs:
+                file_path = object_files_path + '/' + blob.split('/')[-1]
+                blob.download_to_filename(file_path)
+        elif (self.ss_name == 'azure_store'):
```

Comment:
I haven't been following along with the implementation of Azure or GCP but how does Azure store handle the prefix for grouping test files together? It doesn't seem to be used here when you download the objects.

Reply:
Good question! For GCP, the prefix grouping is done through the list_blobs function being able to take in a prefix parameter and for Azure, the prefix grouping is done through list_blobs with `name_starts_with` argument.